ext/mysql2/violite.h

```diff
@@ -0,0 +1,280 @@
+/* Copyright (c) 2000, 2015, Oracle and/or its affiliates. All rights
```

Comment:
Looking more closely, it's a little worse than just defining the one `has_data` method, we're actually becoming aware of the `st_vio` structure in order to call one of its member function pointers. I'm concerned that this isn't part of the "public API" and would be subject to change across MySQL versions.